# metabuli-analysis
Scripts used for database creation and classification of each tool in benchmarks.

Scripts and raw values for plotting can be found in [metabuli-plots](https://github.com/jaebeom-kim/metabuli-plots).

For the hybrid approach, which runs DNA tool first and AA tool for unclassified reads, naming `ensemble` or `hybrid` is used. Both are the same.

## How to measure performance of tools
The crieteria for true positive and false positive and the formula for calculating precision, recall, and F1 score are defined in the manuscript. 
### Prokaryote benchmarks
- Use `grade` module of `metabuli`
```
metabuli grade <list of classification result files> <list of answer files> <taxonomy dir> --test-type gtdb [options]
    
    options:
    --test-rank: csv of ranks to test (default: "class,order,family,genus,species")
    --readid-col: column number of read ID in a read-by-read classification result
    --taxid-col: column number of tax ID in a read-by-read classification result
    --threads: number of threads to use (default: all)
```
* `<list of classification files>`: A list of files containing read-by-read classification results. One file per line.
* `<list of answer files>`: A list of files containing answers (assembly accession 2 taxonomy ID mapping files). The order of files should match the order of classification files. One file per line. 
* `<taxonomy dir>`: A directory containing taxonomy files. The directory should contain `merged.dmp`, `nodes.dmp`, and `names.dmp` files.

### SARS-CoV-2 benchmarks
-   We inspected reports of each tool and manually extracted the number of classifications to each SARS-CoV-2 variant for inclusion test and to Sarbecovirus for exclusion test.
-   For tools that do not make a report but only read-by-read classification results, we used `metabuli binning2report` to generate a report.
```
metabuli binning2report <i:Binning Result> <o:OUT DIR> <o:JOB ID> <i: TAXONOMY DIR>
```
* `<i:Binning Result>`: A file containing read-by-read classification results.
* `<o:OUT DIR>`: A directory to save the report.
* `<o:JOB ID>`: A job ID to be used to name the report.
* `<i: TAXONOMY DIR>`: A directory containing taxonomy files. The directory should contain `merged.dmp`, `nodes.dmp`, and `names.dmp` files.

### CAMI2 benchmarks
- Use `grade` module of `metabuli`

```
metabuli grade <list of classification result files> <list of answer files> <taxonomy dir> --test-type cami [options]
    
    options:
    --test-rank: csv of ranks to test (default: "class,order,family,genus,species")
    --readid-col: column number of read ID in a read-by-read classification result
    --taxid-col: column number of tax ID in a read-by-read classification result
    --threads: number of threads to use (default: all)
```
* `<list of classification files>`: A list of files containing read-by-read classification results. One file per line.
* `<list of answer files>`: A list of files containing answers (provided CAMI2). The order of files should match the order of classification files. One file per line. 
* `<taxonomy dir>`: A directory containing taxonomy files. The directory should contain `merged.dmp`, `nodes.dmp`, and `names.dmp` files.

### Grade by clade size
- Use 'gradeByCladeSize' module of `metabuli`
```
metabuli gradeByCladeSize <i:result list> <i:answer sheet> <reference assembly list> <i:taxonomy Dir> --test-rank subspecies --clade-rank species
```
* `<i:result list>`: A list of files containing read-by-read classification results. One file per line.
* `<i:answer sheet>`: A file containing answers (assembly accession 2 taxonomy ID mapping files). One file is used for all classification results.
* `<reference assembly list>`: A list of reference assemblies. One file per line. It is used to calculate clade size of database.
* `<i:taxonomy Dir>`: A directory containing taxonomy files. The directory should contain `merged.dmp`, `nodes.dmp`, and `names.dmp` files.


## Metamaps without EM steps
- Use `mapping2taxon` module of `metabuli`
- It takes a mapping result of Metamaps and classifies each read to taxon with the highest mapping quality. If ties occur, it classifies the read to the LCA of the tied taxa.
```
metabuli <i:mapping file> <i: taxonomy directory>
```
* `<i:mapping file>`: mapping file generated by metamaps
* `<i: taxonomy directory>`: A directory containing taxonomy files. The directory should contain `merged.dmp`, `nodes.dmp`, and `names.dmp` files.
